### PR TITLE
Carry honeyalb integration work

### PR DESCRIPTION
**Pull request by nathanleclaire**
All code review feedback should be addressed and almost all instances of elbv2 have been replaced with 'alb' - @maplebed PTAL.

### CODE REVIEW

File: cmd/honeyalb/main.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+func init() {
+	// set the version string to our desired format
+	if BuildID == "" {
+		versionStr = "dev"
+	} else {
+		versionStr = BuildID
+	}
+
+	// init libhoney user agent properly
+	libhoney.UserAgentAddition = "honeyelb/" + versionStr
```
Comment (maplebed) [inline cmd/honeyalb/main.go]: honeyalb

File: cmd/honeyalb/main.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+func cmdELB(args []string) error {
```
Comment (maplebed) [inline cmd/honeyalb/main.go]: cmdALB

File: cmd/honeyalb/main.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+				}
+				logrus.Info("State tracking with high availability enabled - using DynamoDB")
+			} else {
+				stater = state.NewFileStater(opt.StateDir, logbucket.AWSElasticLoadBalancingV2, opt.BackfillHr)
+				logrus.Info("State tracking enabled - using local file system.")
+			}
+			logrus.WithField("hours", time.Duration(opt.BackfillHr)*time.Hour).Debug("Backfill will be")
+
+			// TODO: Switch either publisher or parser?
+			// TODO: REMOVE
```
Comment (maplebed) [inline cmd/honeyalb/main.go]: is this TODO relevant? Could you say what should be REMOVEd?

File: cmd/honeyalb/main.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+	formatter := &logrus.TextFormatter{
+		FullTimestamp: true,
+	}
+	logrus.SetFormatter(formatter)
+
+	logrus.WithField("version", BuildID).Debug("Program starting")
+
+	if opt.Dataset == "aws-$SERVICE-access" {
+		opt.Dataset = "aws-elb-access"
```
Comment (maplebed) [inline cmd/honeyalb/main.go]: alb

File: cmd/honeyalb/main.go [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+	}
+
+	if _, err := os.Stat(opt.StateDir); os.IsNotExist(err) {
+		logrus.WithField("dir", opt.StateDir).Fatal("Specified state directory does not exist")
+	}
+
+	if opt.Version {
+		fmt.Println("honeyelb version", versionStr)
```
Comment (maplebed) [inline cmd/honeyalb/main.go]: alb

File: cmd/honeyalb/main.go [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+	}
+
+	if len(args) == 0 {
+		fmt.Fprintln(os.Stderr, `Usage: `+os.Args[0]+` [--flags] [ls|ingest] [ALB names...]
+
+Use '`+os.Args[0]+` --help' to see available flags.`)
+		os.Exit(1)
+	}
+
+	if err := cmdELB(args); err != nil {
```
Comment (maplebed) [inline cmd/honeyalb/main.go]: cmdALB

File: state/state.go [hunk 7]
```diff
@@ -44,14 +44,12 @@ type DownloadedObject struct {
...
 	Session          *session.Session
-	Service          string
 	BackfillInterval time.Duration
 }
 
-func NewDynamoDBStater(session *session.Session, service string, backfillHrs int) (*DynamoDBStater, error) {
+func NewDynamoDBStater(session *session.Session, backfillHrs int) (*DynamoDBStater, error) {
 	stater := &DynamoDBStater{
 		Session:          session,
-		Service:          service,
```
Comment (maplebed) [inline state/state.go]: I don't understand the connection between this change and the ALB stuff.  Is it going along for the ride? Is Service not useful?

Comment (nathanleclaire) [reply state/state.go]: Going along for the ride, though I can split it out if you want. I noticed `Service` wasn't actually used anywhere in the `DynamoDBStater`, so passing it is not needed.

File: cmd/honeyalb/main.go (hunk 4 above)
Comment (nathanleclaire) [reply cmd/honeyalb/main.go]: We had decided sending ALB and ELB to the same place is OK [here](https://github.com/honeycombio/honeyaws/pull/33#discussion_r169720980) - do you want to stick with that (same dataset)? I don't really have a preference, though I do find not having a `aws-alb-access` a bit confusing as an end user.

**Comment by christineyen** [comment]
🎉

are we now in need of a new logo on honeycomb.io/docs/ ? or a new section (cc @feorlen)? what happens when a new integration gets shipped?

**Comment by feorlen** [comment]
New docs has an Integrations section under Getting Data Into Honeycomb. There are sub-sections under that for various types.

I presume this would go in the same section as the existing [ELB](https://honeycomb.io/docs/connect/aws-elastic-load-balancer/) page?

**Pull request merged**